Fix: focus first active input/widget on load, page navigation, user-added repeat

eyelidlessness opened the pull request:
Fixes enketo/enketo-express#277. The tests aren't exhaustive e.g no tests for non-user-added repeats, how relevance impacts determination of whether a field is active. I'd be happy to add any test scenarios but wanted to get this into review first. It's also probably not an exhaustive fix. I hadn't originally caught the select toggle button case, nor the readonly cases (which is why they're used so much in the tests), and I would be surprised if there aren't other edge cases I haven't encountered.

eyelidlessness committed:
Fix: focus first active input/widget on load, page navigation, user-added repeat

eyelidlessness committed:
Additional focus fixes

- Don't infinitely recurse on TOC navigation
- Don't infinitely recurse on navigation to first/last page
- Don't try to focus when no form control is found

Also includes a test for focus on newly relevant control. Worth noting that this does _not_ solve the same case when navigating by tab key in a non-pages version of the same form.

lognaturel commented:
It's a little bit surprising to me that the first question in the new repeat jumps to the top of the screen but I do think it's a big improvement for keyboard navigation. Is the jump intentional? Is it straightforward to jump/not jump or is that currently a part of what "focus" means?

eyelidlessness committed:
Only scroll in `goToTarget` if necessary

In most cases the browser will scroll to the focused element automatically, but this will fallback to scrolling to the question/target if that doesn't occur.
